### PR TITLE
Network resiliency fixes

### PR DESCRIPTION
**Pull request by cbommas**
*Description of changes:*

- Network resiliency fixes 
- version bump 
- updated dependencies on the latest version of the AWS SDK
- updated tests
- more code clean up and logs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AppSyncCustomNetworkInvoker.java [hunk 1]
```diff
@@ -120,11 +121,15 @@ public void run() {
...
-                                           persistentMutationsCallback.onFailure(new PersistentMutationsError(persistentOfflineMutationObject.responseClassName,
-                                                   persistentOfflineMutationObject.recordIdentifier, new ApolloNetworkException("S3 upload failed.", new IllegalArgumentException("S3ObjectManager not provided."))));
+                                           //Invoke callback
+                                           if (persistentMutationsCallback != null ) {
+                                               persistentMutationsCallback.onFailure(new PersistentMutationsError(persistentOfflineMutationObject.responseClassName,
+                                                       persistentOfflineMutationObject.recordIdentifier, new ApolloNetworkException("S3 upload failed.", new IllegalArgumentException("S3ObjectManager not provided."))));
+                                           }
+                                           //Remove mutation from queue and mark state as completed.
+                                           setMutationAsCompleted(persistentOfflineMutationObject.recordIdentifier);
```
Comment (rohandubal) [inline aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AppSyncCustomNetworkInvoker.java]: Consider renaming setMutationAsCompleted to indicate that this means a mutation execution is completed.

File: aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AppSyncCustomNetworkInvoker.java [hunk 2]
```diff
@@ -156,12 +161,36 @@ public String region() {
                                                    return persistentOfflineMutationObject.region;
                                                }
                                            });
-                                       } catch (Exception e) {
-                                           persistentMutationsCallback.onFailure(new PersistentMutationsError(persistentOfflineMutationObject.responseClassName,
-                                                   persistentOfflineMutationObject.recordIdentifier, new ApolloNetworkException("S3 upload failed.", e)));
+                                       }
+                                       catch (AmazonClientException e) {
+                                           if (e.isRetryable()) {
```
Comment (rohandubal) [inline aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AppSyncCustomNetworkInvoker.java]: Can we investigate if we can find underlying cause? like IOException?

File: aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AppSyncCustomNetworkInvoker.java (hunk 1 above)
Comment (cbommas) [reply aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AppSyncCustomNetworkInvoker.java]: Done.

File: aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AppSyncCustomNetworkInvoker.java (hunk 2 above)
Comment (cbommas) [reply aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AppSyncCustomNetworkInvoker.java]: Good call. Checking specifically for IOException and only retrying in that case.

**Pull request merged**